### PR TITLE
frontend: Redesign details view

### PR DESCRIPTION
**Pull request by cyberphantom52**
# Description
This MR aims to adapt the following mockup by @jannuary :
197234722-5feb814c-811e-48a4-80e9-3db1a42d14e1

## Type of change
- [X] UI Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

File: bottles/frontend/ui/details-bottle.blp [hunk 1]
```diff
@@ -230,6 +215,142 @@ template DetailsBottle : .AdwPreferencesPage {
...
+          ]
+        }
+      }
+    }
+    
+    .AdwPreferencesRow bottom_bar {
+      activatable: false;      
+      
+      Box {
+        height-request:45;
```
Comment (jannuary) [inline bottles/frontend/ui/details-bottle.blp:238]: don't do this

File: bottles/frontend/ui/details-bottle.blp [hunk 2]
```diff
@@ -230,6 +215,142 @@ template DetailsBottle : .AdwPreferencesPage {
...
+        
+        Button add_shortcuts {
+          hexpand: true;
+          has-frame: false;
+          focusable: false;
```
Comment (jannuary) [inline bottles/frontend/ui/details-bottle.blp:243]:
```        has-frame: false;
          focusable: false;```
          
these are unneccessary and break keyboard nav

File: bottles/frontend/ui/details-bottle.blp [hunk 3]
```diff
@@ -230,6 +215,142 @@ template DetailsBottle : .AdwPreferencesPage {
...
+            }
+          
+            Separator {
+              styles [
+                "spacer"
+              ]
+            }
+
+            Label {
+              label: _("Add Programs...");
```
Comment (jannuary) [inline bottles/frontend/ui/details-bottle.blp:281]:
```suggestion
              label: _("Add Programs…");
```

File: bottles/frontend/ui/details-bottle.blp [hunk 4]
```diff
@@ -230,6 +215,142 @@ template DetailsBottle : .AdwPreferencesPage {
...
+            }
+            
+            Separator {
+              styles [
+                "spacer"
+              ]
+            }
+            
+            Label {
+              label: _("Add Shortcuts...");
```
Comment (jannuary) [inline bottles/frontend/ui/details-bottle.blp:257]:
```suggestion
              label: _("Add Shortcuts…");
```

File: bottles/frontend/ui/details-bottle.blp [hunk 5]
```diff
@@ -230,6 +215,142 @@ template DetailsBottle : .AdwPreferencesPage {
...
+
+          styles [
+            "dim-label",
+          ]
+        }
+      }
+    }
+    
+    .AdwPreferencesRow bottom_bar {
+      activatable: false;
```
Comment (jannuary) [inline bottles/frontend/ui/details-bottle.blp:235]: activatable is false by default afaik

Comment (jannuary) [reply bottles/frontend/ui/details-bottle.blp:235]: this should likely have focusable: false though

File: bottles/frontend/ui/details-bottle.blp [hunk 6]
```diff
@@ -230,6 +215,142 @@ template DetailsBottle : .AdwPreferencesPage {
...
+
+        Image {
+          icon-name: "go-next-symbolic";
+        }
+      }
+
+      .AdwActionRow row_snapshots {
+        activatable: true;
+        title: _("Snapshots");
+        subtitle: _("Imagine windows restore points, but better.");
```
Comment (jannuary) [inline bottles/frontend/ui/details-bottle.blp:339]: not helpful nor descriptive :)

File: bottles/frontend/ui/details-bottle.blp [hunk 7]
```diff
@@ -230,6 +215,142 @@ template DetailsBottle : .AdwPreferencesPage {
...
+
+        Image {
+          icon-name: "go-next-symbolic";
+        }
+      }
+
+      .AdwActionRow row_taskmanager {
+        activatable: true;
+        title: _("Taskmanager");
```
Comment (jannuary) [inline bottles/frontend/ui/details-bottle.blp:348]:
```suggestion
        title: _("Task Manager");
```

File: bottles/frontend/ui/details-bottle.blp (hunk 6 above)
Comment (cyberphantom52) [reply bottles/frontend/ui/details-bottle.blp:339]: Texts are not final in this PR yet, i just put whatever for now :)

File: bottles/frontend/ui/details-bottle.blp (hunk 1 above)
Comment (cyberphantom52) [reply bottles/frontend/ui/details-bottle.blp:277]: image
it looks like this without `height-request` and `has-frame` for buttons not set.

Comment (jannuary) [reply bottles/frontend/ui/details-bottle.blp:277]: you replace it with .flat style and the css in discord

File: bottles/frontend/ui/details-bottle.blp (hunk 2 above)
Comment (cyberphantom52) [reply bottles/frontend/ui/details-bottle.blp:280]: in my testing, `has-frame` is definitely required. `focusable` has been mistakenly pushed, i was experimenting with it earlier,  removed it :)

File: bottles/frontend/ui/details-bottle.blp (hunk 5 above)
Comment (cyberphantom52) [reply bottles/frontend/ui/details-bottle.blp:274]: No, its true by default, need to be false else the whole bottom bar is highlighted on hover.
`focusable` does nothing as per my testing

Comment (cyberphantom52) [reply bottles/frontend/ui/details-bottle.blp:274]: aabc0162f7230c82a404f8da9d709a41f314e791

File: bottles/frontend/ui/details-bottle.blp (hunk 2 above)
Comment (cyberphantom52) [reply bottles/frontend/ui/details-bottle.blp]: 2294c0ddcf2c4d75b9ae36b69247ae5fb6911abc

File: bottles/frontend/ui/details-bottle.blp (hunk 1 above)
Comment (cyberphantom52) [reply bottles/frontend/ui/details-bottle.blp]: a48e8cda046903ca3dfbbafd316bf153c09675f0

File: bottles/frontend/ui/details-bottle.blp (hunk 5 above)
Comment (jannuary) [reply bottles/frontend/ui/details-bottle.blp:274]: > No, its true by default, need to be false else the whole bottom bar is highlighted on hover.

It doesn't seem as such?

File: bottles/frontend/ui/details-bottle.blp (hunk 4 above)
Comment (cyberphantom52) [reply bottles/frontend/ui/details-bottle.blp]: 4f3000e3b77eeb32138e4dbf4be25cec9dadf766

File: bottles/frontend/ui/details-bottle.blp (hunk 3 above)
Comment (cyberphantom52) [reply bottles/frontend/ui/details-bottle.blp]: 4f3000e3b77eeb32138e4dbf4be25cec9dadf766

File: bottles/frontend/ui/details-bottle.blp [hunk 8]
```diff
@@ -269,6 +254,141 @@ template DetailsBottle : .AdwPreferencesPage {
...
+      
+      .AdwActionRow row_programs {
+        activatable: true;
+        title: _("Programs");
+        subtitle: _("Manage installed programs.");
+
+        Image {
+          icon-name: "go-next-symbolic";
+        }
+      }
```
Comment (TheEvilSkeleton) [inline bottles/frontend/ui/details-bottle.blp:371]: We'll have to remove this, as it received some complaints (see https://github.com/bottlesdevs/Bottles/issues/2126 and https://gitlab.gnome.org/Teams/Circle/-/issues/61).

I propose to have the Programs section in the details view the only way to manage programs. That way we won't have duplicated options and everything will be available.

**Review by TheEvilSkeleton — CHANGES_REQUESTED** [review]
For `details.blp`, I can't swipe on my laptop's trackpad to go back.

https://user-images.githubusercontent.com/50847364/198322206-3fae3926-9e51-4e5b-87b9-d75929bf267c.mp4

This is a minor issue, but I'd like to have it fixed here. (Granted, I have no idea how to fix this)

**Comment by jannuary** [comment]
> For `details.blp`, I can't swipe on my laptop's trackpad to go back.
> This is a minor issue, but I'd like to have it fixed here. (Granted, I have no idea how to fix this)

This was removed because it was causing the page to get deleted too soon and is as such unrelated to this PR. Though, tbf, nested swipeable leaflets are finnicky anyway:
image

**Comment by TheEvilSkeleton** [comment]
No worries then.

File: bottles/frontend/ui/details-bottle.blp (hunk 8 above)
Comment (cyberphantom52) [reply bottles/frontend/ui/details-bottle.blp]: 92200582ef83a388ac3f3565fb158cc8a478c5fa

**Review by jannuary — APPROVED** [review]

File: bottles/frontend/ui/details-bottle.blp [hunk 9]
```diff
@@ -40,6 +40,17 @@ Popover pop_context {
     Separator {
     }
 
+    .GtkModelButton btn_toggle_removed {
+      text: _("Toggle Hidden");
```
Comment (TheEvilSkeleton) [inline bottles/frontend/ui/details-bottle.blp:44]: It's difficult to tell whether something is toggled or not without a clear indication. Should this toggle be renamed to "Show Hidden", when it's toggled off, and "Hide Hidden", when it's toggled on?

Comment (cyberphantom52) [reply bottles/frontend/ui/details-bottle.blp:44]: "Hide Hidden" is kinda whacky

Comment (cyberphantom52) [reply bottles/frontend/ui/details-bottle.blp]: 157305ea16772f51af8fb2761ed0d5b88cf6c5c0

File: bottles/frontend/ui/details-bottle.blp [hunk 10]
```diff
@@ -40,6 +40,17 @@ Popover pop_context {
     Separator {
     }
 
+    .GtkModelButton btn_toggle_removed {
+      text: _("Show Hidden");
```
Comment (TheEvilSkeleton) [inline bottles/frontend/ui/details-bottle.blp:44]:
```suggestion
      text: _("Show Hidden Programs");
```

File: bottles/frontend/views/bottle_details.py [hunk 11]
```diff
@@ -205,15 +239,115 @@ def set_config(self, config):
...
+        self.group_programs.add(self.bottom_bar) # Add the bottom_bar back to the bottom
+
+    def __toggle_removed(self, widget=False):
+        """
+        This function toggle the show_hidden variable.
+        """
+        if self.show_hidden:
+            self.btn_toggle_removed.set_property('text', _("Show Hidden"))
+        else:
+            self.btn_toggle_removed.set_property('text', _("Hide Hidden"))
```
Comment (TheEvilSkeleton) [inline bottles/frontend/views/bottle_details.py:345]:
```suggestion
            self.btn_toggle_removed.set_property('text', _("Show Hidden Programs"))
        else:
            self.btn_toggle_removed.set_property('text', _("Hide Hidden Programs"))
```

**Review by TheEvilSkeleton — CHANGES_REQUESTED** [review]
"[...] Hidden Programs" is clearer.

**Review by TheEvilSkeleton — APPROVED** [review]

**Pull request merged**